--- advancecomp-2.6/Makefile.am.orig	2023-10-10 17:01:16.000000000 +0200
+++ advancecomp-2.6/Makefile.am	2024-04-17 18:59:18.464658604 +0200
@@ -21,34 +21,10 @@ bin_PROGRAMS = advzip advpng advmng advd
 	7z/WindowIn.cc \
 	7z/WindowOut.cc
 
-libdeflate_SOURCES = \
-	libdeflate/arm/cpu_features.c \
-	libdeflate/x86/cpu_features.c \
-	libdeflate/adler32.c \
-	libdeflate/crc32.c \
-	libdeflate/deflate_compress.c\
-	libdeflate/deflate_decompress.c \
-	libdeflate/gzip_compress.c \
-	libdeflate/gzip_decompress.c \
-	libdeflate/utils.c \
-	libdeflate/zlib_compress.c \
-	libdeflate/zlib_decompress.c
-
-
-zopfli_SOURCES = \
-	zopfli/blocksplitter.c \
-	zopfli/cache.c \
-	zopfli/deflate.c \
-	zopfli/gzip_container.c \
-	zopfli/hash.c \
-	zopfli/katajainen.c \
-	zopfli/lz77.c \
-	zopfli/squeeze.c \
-	zopfli/tree.c \
-	zopfli/util.c \
-	zopfli/zlib_container.c \
-	zopfli/zopfli.h \
-	zopfli/zopfli_lib.c
+libdeflate_LIBS = -ldeflate
+
+
+zopfli_LIBS = -lzopfli
 
 advzip_SOURCES = \
 	rezip.cc \
@@ -62,8 +38,8 @@ advzip_SOURCES = \
 	snprintf.c \
 	portable.c \
 	$(7z_SOURCES) \
-	$(libdeflate_SOURCES) \
-	$(zopfli_SOURCES)
+	$(nil)
+advzip_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
 
 advpng_SOURCES = \
 	repng.cc \
@@ -79,8 +55,8 @@ advpng_SOURCES = \
 	lib/error.c \
 	lib/snstring.c \
 	$(7z_SOURCES) \
-	$(libdeflate_SOURCES) \
-	$(zopfli_SOURCES)
+	$(nil)
+advpng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
 
 advmng_SOURCES = \
 	remng.cc \
@@ -101,8 +75,8 @@ advmng_SOURCES = \
 	lib/error.c \
 	lib/snstring.c \
 	$(7z_SOURCES) \
-	$(libdeflate_SOURCES) \
-	$(zopfli_SOURCES)
+	$(nil)
+advmng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
 
 advdef_SOURCES = \
 	redef.cc \
@@ -120,8 +93,8 @@ advdef_SOURCES = \
 	lib/png.c \
 	lib/mng.c \
 	$(7z_SOURCES) \
-	$(libdeflate_SOURCES) \
-	$(zopfli_SOURCES)
+	$(nil)
+advdef_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
 
 # Add the .version file in the distribution
 dist-hook:
--- advancecomp-1.23/compress.h.orig	2016-09-26 22:04:03.000000000 +0200
+++ advancecomp-1.23/compress.h	2016-12-18 22:11:26.444429691 +0100
@@ -24,11 +24,11 @@
 #include "7z/7z.h"
 
 extern "C" {
-#include "libdeflate/libdeflate.h"
+#include <libdeflate.h>
 }
 
 extern "C" {
-#include "zopfli/zopfli.h"
+#include <zopfli.h>
 }
 
 #if USE_BZIP2
