+++ advancecomp-2.6/Makefile.am.orig	2024-04-17 18:59:18.464658604 +0200
+++ advancecomp-2.6/Makefile.am	2024-04-17 19:02:19.014563123 +0200
@@ -1,6 +1,10 @@
 bin_PROGRAMS = advzip advpng advmng advdef
 
-7z_SOURCES = \
+lib_LTLIBRARIES = libadv7z.la
+adv7zincludedir = $(includedir)/adv7z
+adv7zinclude_HEADERS = 7z/7z.h
+
+libadv7z_la_SOURCES = \
 	7z/7zdeflate.cc \
 	7z/7zlzma.cc \
 	7z/AriBitCoder.cc \
@@ -37,9 +41,8 @@ advzip_SOURCES = \
 	getopt.c \
 	snprintf.c \
 	portable.c \
-	$(7z_SOURCES) \
 	$(nil)
-advzip_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
+advzip_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS) libadv7z.la
 
 advpng_SOURCES = \
 	repng.cc \
@@ -55,9 +58,8 @@ advpng_SOURCES = \
 	lib/png.c \
 	lib/error.c \
 	lib/snstring.c \
-	$(7z_SOURCES) \
 	$(nil)
-advpng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
+advpng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS) libadv7z.la
 
 advmng_SOURCES = \
 	remng.cc \
@@ -76,9 +78,8 @@ advmng_SOURCES = \
 	lib/mng.c \
 	lib/error.c \
 	lib/snstring.c \
-	$(7z_SOURCES) \
 	$(nil)
-advmng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
+advmng_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS) libadv7z.la
 
 advdef_SOURCES = \
 	redef.cc \
@@ -95,9 +96,8 @@ advdef_SOURCES = \
 	lib/snstring.c \
 	lib/png.c \
 	lib/mng.c \
-	$(7z_SOURCES) \
 	$(nil)
-advdef_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS)
+advdef_LDADD = $(libdeflate_LIBS) $(zopfli_LIBS) libadv7z.la
 
 # Add the .version file in the distribution
 dist-hook:
--- advancecomp-2.6/configure.ac.orig	2023-10-10 17:17:08.000000000 +0200
+++ advancecomp-2.6/configure.ac	2024-04-17 19:07:04.858546549 +0200
@@ -4,6 +4,7 @@ dnl Get version number from git
 m4_define([git_revision], m4_esyscmd_s([./autover.sh]))
 AC_INIT([AdvanceCOMP], [git_revision], [], [], [http://www.advancemame.it])
 AM_INIT_AUTOMAKE([foreign no-dependencies subdir-objects])
+LT_INIT
 AC_CONFIG_SRCDIR([rezip.cc])
 AC_CONFIG_HEADERS([config.h])
 AC_CANONICAL_HOST
